opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -221,7 +223,11 @@ def get_meter(
             )
             return _DefaultMeter(name, version=version, schema_url=schema_url)
 
-        return Meter(
-            InstrumentationInfo(name, version, schema_url),
-            self._measurement_consumer,
-        )
+        with self._meter_lock:
+            info = InstrumentationInfo(name, version, schema_url)
```

Comment:
This is fine but it depends on the specific implementation details of `InstrumentationInfo` (its `hash` method) and it also calls for the creation of a new object every time. The concatenation of `name`, `version` and `schema_url` make also an unique key for `self._meters` that does not require the creation of a new `InstrumentationInfo` object every time.

Reply:
We need to make the object anyway for L230 (see suggested change) so I think it's ok. `InstrumentationInfo.hash()` should be the source of truth here so i prefer this impl over inlining it

One nit though, you can do this instantiation outside of the locked section